### PR TITLE
Handle errors on aggregated nodes

### LINKED ISSUE #46: exceptions on agherant nodes subrequests not handled
steps to reproduce the error:
 - run ```WEBANT_AGHERANT_DESCRIPTIONS=['http://example.org'] agherant```
 - visit ```/agherant/search?q=*:*```

the exception raised is not very clear:
```
Traceback (most recent call last):
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/libreant/libreant/webant/agherant.py", line 18, in search
    books = aggregate(current_app.config['AGHERANT_DESCRIPTIONS'], query)
  File "/home/libreant/libreant/webant/agherant.py", line 39, in aggregate
    clients = map(Client, descriptions)
  File "/home/libreant/libreant/webant/utils.py", line 12, in memoizer
    cache[key] = obj(*args, **kwargs)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/opensearch-0.9.2-py2.7.egg/opensearch/client.py", line 44, in __init__
    self.description = Description(url, self.agent)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/opensearch-0.9.2-py2.7.egg/opensearch/description.py", line 16, in __init__
    self.load(url)
  File "/home/libreant/libreant/ve/lib/python2.7/site-packages/opensearch-0.9.2-py2.7.egg/opensearch/description.py", line 24, in load
    self.dom = parse(urlopen(req))
  File "/usr/lib/python2.7/xml/dom/minidom.py", line 1920, in parse
    return expatbuilder.parse(file)
  File "/usr/lib/python2.7/xml/dom/expatbuilder.py", line 928, in parse
    result = builder.parseFile(file)
  File "/usr/lib/python2.7/xml/dom/expatbuilder.py", line 207, in parseFile
    parser.Parse(buffer, 0)
ExpatError: syntax error: line 1, column 0
```

### PR DESCRIPTION
**Pull request by boyska**
You can test that it works running this

```sh
 WEBANT_AGHERANT_DESCRIPTIONS='["SELF", "http://example.org"]' webant
```

previously, this would have been an error (parsing example.org as an xml).
Now it is not, and it works correctly, aggregating itself.

Closes #46

### CODE REVIEW

File: webant/agherant.py [hunk 1]
```diff
@@ -36,7 +38,13 @@ def autoreplace(description_url):
             return url_for('description', _external=True)
         return description_url
     descriptions = map(autoreplace, descriptions)
-    clients = map(Client, descriptions)
+    clients = []
+    for url in descriptions:
+        try:
+            clients.append(Client(url))
+        except Exception as exc:
+            log.info("Error while retrieving description for '%s'" % url)
```
Comment (ael-code) [inline webant/agherant.py]: reading from [python logger doc](https://docs.python.org/2/library/logging.html#logging.Logger.exception) and [stackoverflow](http://stackoverflow.com/questions/5191830/python-exception-logging)
I think you should write like this:
```python
log.exception("Error while retrieving description for '{}'".format(url))
```

**Comment by boyska** [comment]
thanks, fixed

**Comment by ael-code** [comment]
if i use ```"['SELF','http://example.org']"``` instead of ```'["SELF","http://example.org"]'``` the config is handle as a list of chars "[,',S,E,L...."
Isn't strange?

**Comment by boyska** [comment]
No, it's not. JSON string must be in double quotes.

> A string is a sequence of zero or more Unicode characters, wrapped in double quotes, using backslash escapes ( from www.json.org)

AppConfig tries to parse the variable as json; when it fails, it uses it as a string. Then we're interpreting that string as a list, which makes it "explode" to a list of single characters.

**Comment by ael-code** [comment]
> AppConfig tries to parse the variable as json; when it fails, it uses it as a string. Then we're interpreting that string as a list, which makes it "explode" to a list of single characters.

when we will implement agherant doc ( #23 ), it will be nice to insert an allert to remember this to sysadmins.

**Comment by boyska** [comment]
I think that this can be merged; you didn't said it clearly, but I believe you have tested this and it works (when the value of `WEBANT_AGHERANT_DESCRIPTIONS` is valid, of course!)

**Pull request merged**